README
Business:PostNL
====================

INSTALLATION

To install this module type the following:

   perl Makefile.PL
   make
   make test
   make install

DEPENDENCIES

   Carp
   Class::Accessor::Fast
   List::Util
   Test::Simple
   YAML

COPYRIGHT AND LICENCE

Copyright (C) 2006..2013 by M. Blom

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself, either Perl version 5.8.4 or,
at your option, any later version of Perl 5 you may have available.



Changes
Revision history for Perl module Business::PostNL

0.12 Jan 29 2013
    - bumped the version to make PAUSE happy

0.12 Jan 23 2013
    - renamed from Business::TNTPost::NL to Business::PostNL
      yes, it's amazing how often this company changes names.
    - new rates

0.11_01 Jan 19 2011
    - Fixed errors reported by altering logic and adding registered costs per zone

0.11 Jan 16 2011
    - Fixed rates according to new PDF information

    NOTE:
      - 'receipt' is no longer an option
      - 'priority' is default for all non-NL shipping

0.10 Feb 26 2009
    - Fixed letter rates for RoW1

0.08 Jan 22 2009
    - Fixed countries considered "Europe"

0.06 Dec 24 2008
    - new rates, updated module
    - added more pod
    - using Class::Accessor::Fast

0.04 Feb 15 2008
    - fixed bug in world :: basic :: europe :: large :: standard

0.03 Jan 09 2008
    - fixed bug in world :: register :: world
    - fixed country codes (Greece is back in zone 4)

0.02 Dec 14 2007
    - new rates, updated module

0.01 Dec 31 2006
    - original version, originated from the obsolete Business::TPGPost


lib/Business/PostNL.pm
package Business::PostNL;

use strict;
use Business::PostNL::Data qw/:ALL/;
use Carp;
use List::Util qw/reduce/;

our $VERSION = 0.13;
our $ERROR   = undef;

use base qw/Class::Accessor::Fast/;

BEGIN {
    __PACKAGE__->mk_accessors(
        qw/cost large machine priority receipt register tracktrace weight zone/
    );
}

=pod

=head1 NAME

Business::PostNL - Calculate Dutch (PostNL) shipping costs

=head1 SYNOPSIS

  use Business::PostNL;

  my $tnt = Business::PostNL->new();
     $tnt->country('DE');
     $tnt->weight('534');
     $tnt->large(1);
     $tnt->tracktrace(1);
     $tnt->register(1);

  my $costs = $tnt->calculate or die $Business::PostNL::ERROR;


or

  use Business::PostNL;

  my $tnt = Business::PostNL->new();
  my $costs = $tnt->calculate(
                  country    =>'DE',
                  weight     => 534,
                  large      => 1,
                  tracktrace => 1,
                  register   => 1,
              ) or die $Business::PostNL::ERROR;

=head1 DESCRIPTION

This module calculates the shipping costs for the Dutch PostNL,
based on country, and weight etc.

The shipping cost information is based on 'Posttarieven 2013'.

It returns the shipping costs in euro or undef (which usually means
the parcel is heavier than the maximum allowed weight; check
C<$Business::PostNL::ERROR>).

=head2 METHODS

The following methods can be used

=head3 new

C<new> creates a new C<Business::PostNL> object. No more, no less.

=cut

sub new {
    my ( $class, %parameters ) = @_;
    my $self = bless( {}, ref($class) || $class );
    return $self;
}

=pod

=head3 country

Sets the country (ISO 3166, 2-letter country code) and returns the
zone number used by PostNL (or 0 for The Netherlands (NL)).

This value is mandatory for the calculations.

Note that the reserved IC has been used for the Canary Islands. This
has not been adopted by ISO 3166 yet. The Channel Islands are completely
ignored due to a lack of a code.

=cut

sub country {
    my ( $self, $cc ) = @_;

    if ($cc) {
        my $zones = Business::PostNL::Data::zones();
        $self->zone( defined $zones->{$cc} ? $zones->{$cc} : '4' );
    }

    return $self->zone;
}

=pod

=head3 calculate

Method to calculate the actual shipping cost based on the input (see
methods above). These options can also be passed straight in to this method
(see L<SYNOPSIS>).

Two settings are mandatory: country and weight. The rest are given a
default value that will be used unless told otherwise.

Returns the shipping costs in euro, or undef (see $Business::PostNL::ERROR
in that case).

=cut

sub calculate {
    my ( $self, %opt ) = @_;

    # Set the options
    for ( qw/country weight large tracktrace register machine/) {
        $self->$_( $opt{$_} ) if ( defined $opt{$_} );
    }

    croak "Not enough information!"
      unless ( defined $self->zone && defined $self->weight );

    # > 2000 grams automatically means 'tracktrace'
    $self->tracktrace(1) if ( $self->weight > 2000 );

    # Fetch the interesting table
    my $ref = _pointer_to_element( table(), $self->_generate_path );
    my $table = $$ref;

    my $highest = 0;
    foreach my $key ( keys %{$table} ) {
        my ( $lo, $hi ) = split ',', $key;
        $highest = $hi if ( $hi > $highest );
        if ( $self->weight >= $lo && $self->weight <= $hi ) {
            $self->cost( $table->{$key} );
            last;
        }
    }
    $ERROR = $self->weight - $highest . " grams too heavy (max: $highest gr.)"
      if ( $highest < $self->weight );

    return ( $self->cost ) ? sprintf( "%0.2f", $self->cost ) : undef;
}

=pod

=head3 _generate_path

Internal method to create the path to walk through the pricing table.
Don't call this, use L<calculate> instead.

=cut

sub _generate_path {
    my $self = shift;

    my @p;

    if ( $self->zone ) {
        push @p, 'world';           # world

        if( $self->large ) {
            push @p, 'large',       # w/large
                     'zone',        # w/large/zone
                     $self->zone;   # w/large/zone/[1..4]
            push @p,
                ( $self->machine )
                ? 'machine'         # w/large/zone/[1..4]/machine
                : 'stamp';          # w/large/zone/[1..4]/stamp

            push @p,
                ( $self->register )
                ? 'register'        # w/large/zone/[1..4]/(m|s)/register
                : ( $self->tracktrace )
                  ? 'tracktrace'    # w/large/zone/[1..4]/(m|s)/tracktrace
                  : 'normal';       # w/large/zone/[1..4]/(m|s)/normal

        }
        else {
            push @p, 'small';       # w/small
            push @p,
                ( $self->zone < 4 )
                ? 'europe'          # w/small/europe
                : 'world';          # w/small/world
            push @p,
                ( $self->machine )
                ? 'machine'         # w/small/(e|w)/machine
                : 'stamp';          # w/small/(e|w)/stamp
            push @p,
                ( $self->register )
                ? 'register'        # w/small/(e|w)/(m|s)/register
                : 'normal';         # w/small/(e|w)/(m|s)/normal
        }
    }
    else {
        push @p, 'netherlands';                 # netherlands
        if ( $self->register ) {
            push @p, 'register';                # n/register
        }
        else {
            push @p, ( $self->large )           # n/(large|small)
              ? 'large'
              : 'small';
        }
        push @p,
            ( $self->machine )
            ? 'machine'                         # n/(r|l|s)/machine
            : 'stamp';                          # n/(r|l|s)/stamp
    }
    #print (join " :: ", @p), "\n";
    return @p;
}

=pod

=head3 _pointer_to_element

Blame L<merlyn> for this internal method. It's using L<List::Util>
to "grep" the information needed.

Don't call this, use L<calculate> instead.

=cut

sub _pointer_to_element {                       # Thanks 'merlyn'!
    require List::Util;
    return List::Util::reduce( sub { \( $$a->{$b} ) }, \shift, @_ );
}

=pod

=head3 weight

Sets and/or returns the weight of the parcel in question in grams.

This value is mandatory for the calculations.

=head3 large

Sets and/or returns the value of this option. Defaults to undef (meaning:
the package will fit through the mail slot).

=head3 priority [DEPRECATED]

PostNL still requires you to put a priority sticker on your letters and
parcels, but this seems to be solely for speed of delivery. I couldn't
find any price difference, hence this setting is ignored from now on and
only here for backwards compatability.

=head3 tracktrace

Sets and/or returns the value of this options. Defaults to undef (meaning:
no track & trace feature wanted). When a parcel destined for abroad
weighs over 2 kilograms, default is 1, while over 2kg it's not even
optional anymore.

=head3 register

Sets and/or returns the value of this options. Defaults to undef (meaning:
parcel is not registered (Dutch: aangetekend)).

=head3 receipt [DEPRECATED]

No longer an option, solely here for backwards compatibility.


=head3 machine

Sets and/or returns the value of this options. Defaults to undef (meaning:
stamps will be used, not the machine (Dutch: frankeermachine)).

Only interesting for destinies within NL. Note that "Pakketzegel AVP"
and "Easystamp" should also use this option.

=head1 BUGS

Please do report bugs/patches to
L<http://rt.cpan.org/Public/Dist/Display.html?Name=Business-PostNL>

=head1 CAVEAT

The Dutch postal agency (PostNL) uses many, many, many various ways
for you to ship your parcels. Some of them are included in this module,
but a lot of them not (maybe in the future? Feel free to patch ;-)

=head1 AUTHOR

Menno Blom,
E<lt>blom@cpan.orgE<gt>

=head1 COPYRIGHT

This program is free software; you can redistribute
it and/or modify it under the same terms as Perl itself.

The full text of the license can be found in the
LICENSE file included with this module.

=head1 SEE ALSO

L<http://www.postnl.nl/>,
L<http://www.postnl.nl/voorthuis/images/PostNL-Tarievenmap-2013_tcm213-513266.pdf>,
L<http://www.iso.org/iso/en/prods-services/iso3166ma/index.html>

=cut

1;


lib/Business/PostNL/Data.pm
package Business::PostNL::Data;
use strict;
use base 'Exporter';
use vars qw($VERSION @EXPORT @EXPORT_OK %EXPORT_TAGS);

use Carp;
use YAML;

$VERSION   = 0.13;
@EXPORT    = qw();
@EXPORT_OK = qw(zones table);
%EXPORT_TAGS = ("ALL" => [@EXPORT_OK]);

=pod

=head1 NAME

Business::PostNL::Data - Shipping cost data for Business::PostNL

=head1 DESCRIPTION

Data module for Business::PostNL containing shipping cost
information, country zones etc.

Nothing to see here, the show is over, move along please.

=head2 METHODS

The following methods are used and can be exported

=head3 zones

Returns a hashref with country and zone numbers

=cut

sub zones {
   my %zones = (
      0 => [ qw(NL) ],                                           # NL
      1 => [ qw(BE LU DK DE FR IT AT ES GB SE) ],                # EU 1
      2 => [ qw(BG EE FI HU IE LV LT PL PT RO SI SK CZ) ],       # EU 2
      3 => [ qw(AL AD BA IC CY FO GI GR GL IS HR LI MK MD MT
                ME NO UA SM RS TR VA BY CH) ],                   # EU 3
   );
   my %z;
   foreach my $key (keys %zones) {
      foreach my $val (@{$zones{$key}}) {
         $z{$val} = $key;
      }
   }
   return \%z;
}

=pod

=head3 table

This method contains the heart of this module, the lookup table

=cut

sub table {
my $table = Load(<<'...');
---
# Netherlands
netherlands:
  # Letters (brievenbuspost)
  small:
    stamp:
      '0,20': 0.54
      '21,50': 1.08
      '51,100': 1.62
      '101,250': 2.16
      '251,500': 2.70
      '501,2000': 2.70
    machine:
      '0,20': 0.47
      '21,50': 0.89
      '51,100': 1.41
      '101,250': 1.88
      '251,500': 2.35
      '501,2000': 2.82
  # Parcels (paketten)
  large:
    stamp:
      '2000,10000': 6.75
      '10001,30000': 12.40
    machine:
      '2000,10000': 6.75
  # Register (aangetekend)
  register:
    stamp:
      '0,2000': 7.70
      '2001,10000': 8.05
      '100001,30000': 13.70
    machine:
      '0,2000': 7.47
      '2001,10000': 6.75
# Outside of the Netherlands
world:
  # letters
  small:
    # Within Europe (EU1 & EU2 & EU3)
    europe:
      stamp:
        normal:
          '0,20': 0.90
          '21,50': 1.80
          '51,100': 2.70
          '101,250': 4.50
          '251,500': 7.20
          '501,2000': 9.00
        register:
          '0,2000': 9.50
      machine:
        normal:
          '0,20': 0.87
          '21,50': 1.75
          '51,100': 2.62
          '101,250': 4.37
          '251,500': 6.98
          '501,2000': 8.73
        register:
          '0,2000': 9.22
    # Outside Europe
    world:
      stamp:
        normal:
          '0,20': 0.95
          '21,50': 1.90
          '51,100': 2.85
          '101,250': 4.75
          '251,500': 9.50
          '500,2000': 12.35
        register:
          '0,2000': 16.00
      machine:
        normal:
          '0,20': 0.92
          '21,50': 1.84
          '51,100': 2.75
          '101,250': 4.61
          '251,500': 9.22
          '501,2000': 11.98
        register:
          '0,2000': 15.52
  # parcels
  large:
    zone:
      # EU1
      1:
        stamp:
          normal:
            '0,2000': 9.00
          tracktrace:
            '0,2000': 13.00
            '2001,5000': 19.50
            '5001,10000': 25.00
            '10001,20000': 34.00
            '20001,30000': 41.15
          register:
            '0,2000': 14.30
            '2001,5000': 20.80
            '5001,10000': 26.30
            '10001,20000': 35.30
            '20001,30000': 42.45
        machine:
          normal:
            '0,2000': 8.73
          tracktrace:
            '0,2000': 13.00
            '2001,5000': 19.50
            '5001,10000': 25.00
            '10001,20000': 34.00
          register:
            '0,2000': 14.30
            '2001,5000': 20.80
            '5001,10000': 26.30
            '10001,20000': 35.30
      # EU2
      2:
        stamp:
          normal:
            '0,2000': 10.80
          tracktrace:
            '0,2000': 18.50
            '2001,5000': 25.00
            '5001,10000': 31.00
            '10001,20000': 40.00
            '20001,30000': 48.40
          register:
            '0,2000': 19.80
            '2001,5000': 26.30
            '5001,10000': 32.30
            '10001,20000': 41.30
            '20001,30000': 49.70
        machine:
          normal:
            '0,2000': 10.48
          tracktrace:
            '0,2000': 18.50
            '2001,5000': 25.00
            '5001,10000': 31.00
            '10001,20000': 40.00
          register:
            '0,2000': 19.80
            '2001,5000': 26.30
            '5001,10000': 32.30
            '10001,20000': 41.30
      # EU3
      3:
        stamp:
          normal:
            '0,2000': 11.70
          tracktrace:
            '0,2000': 19.30
            '2001,5000': 26.30
            '5001,10000': 32.30
            '10001,20000': 42.30
          register:
            '0,2000': 20.60
            '2001,5000': 27.60
            '5001,10000': 33.60
            '10001,20000': 43.60
        machine:
          normal:
            '0,2000': 11.35
          tracktrace:
            '0,2000': 19.30
            '2001,5000': 26.30
            '5001,10000': 32.30
            '10001,20000': 42.30
          register:
            '0,2000': 20.60
            '2001,5000': 27.60
            '5001,10000': 33.60
            '10001,20000': 43.60
      # RoW2
      4:
        stamp:
          normal:
            '0,2000': 18.05
          tracktrace:
            '0,2000': 24.30
            '2001,5000': 34.30
            '5001,10000': 58.30
            '10001,20000': 105.30
          register:
            '0,2000': 25.60
            '2001,5000': 35.60
            '5001,10000': 59.60
            '10001,20000': 106.60
        machine:
          normal:
            '0,2000': 17.51
          tracktrace:
            '0,2000': 24.30
            '2001,5000': 34.30
            '5001,10000': 58.30
            '10001,20000': 105.30
          register:
            '0,2000': 25.60
            '2001,5000': 35.60
            '5001,10000': 59.60
            '10001,20000': 106.60
...
   return $table;
}

=pod

=head1 AUTHOR

Menno Blom,
E<lt>blom@cpan.orgE<gt>,
L<http://menno.b10m.net/perl/>

=head1 COPYRIGHT

This program is free software; you can redistribute
it and/or modify it under the same terms as Perl itself.

The full text of the license can be found in the
LICENSE file included with this module.

=head1 SEE ALSO

L<Business::PostNL>,
L<http://www.postnl.nl/>

=cut

1;


lib/Business/TNTPost/NL.pm
package Business::TNTPost::NL;

use strict;
use Business::TNTPost::NL::Data qw/:ALL/;
use Carp;
use List::Util qw/reduce/;

our $VERSION = 0.10;
our $ERROR   = undef;

use base qw/Class::Accessor::Fast/;

BEGIN {
    __PACKAGE__->mk_accessors(
        qw/cost large machine priority receipt register tracktrace weight zone/
    );
}

=pod

=head1 NAME

Business::TNTPost::NL - Calculate Dutch (TNT Post) shipping costs

=head1 SYNOPSIS

  use Business::TNTPost::NL;

  my $tnt = Business::TNTPost::NL->new();
     $tnt->country('DE');
     $tnt->weight('534');
     $tnt->large(1);
     $tnt->priority(1);
     $tnt->tracktrace(1);
     $tnt->register(1);
     $tnt->receipt(1);

  my $costs = $tnt->calculate or die $Business::TNTPost::NL::ERROR;
  

or

  use Business::TNTPost::NL;

  my $tnt = Business::TNTPost::NL->new();
  my $costs = $tnt->calculate(
                  country    =>'DE', 
                  weight     => 534, 
                  large      => 1, 
                  tracktrace => 1,
                  register   => 1,
                  receipt    => 1
              ) or die $Business::TNTPost::NL::ERROR;

=head1 DESCRIPTION

This module calculates the shipping costs for the Dutch TNT Post,
based on country, weight and priority shipping (or not), etc.

The shipping cost information is based on 'Tarieven Januari 2009'.

It returns the shipping costs in euro or undef (which usually means
the parcel is heavier than the maximum allowed weight; check
C<$Business::TNTPost::NL::ERROR>).

=head2 METHODS

The following methods can be used

=head3 new

C<new> creates a new C<Business::TNTPost::NL> object. No more, no less.

=cut

sub new {
    my ( $class, %parameters ) = @_;
    my $self = bless( {}, ref($class) || $class );
    return $self;
}

=pod

=head3 country

Sets the country (ISO 3166, 2-letter country code) and returns the
zone number used by TNT Post (or 0 for The Netherlands (NL)).

This value is mandatory for the calculations.

Note that the reserved IC has been used for the Canary Islands. This
has not been adopted by ISO 3166 yet. The Channel Islands are completely
ignored due to a lack of a code.

=cut

sub country {
    my ( $self, $cc ) = @_;

    if ($cc) {
        my $zones = Business::TNTPost::NL::Data::zones();
        $self->zone( defined $zones->{$cc} ? $zones->{$cc} : '4' );
    }

    return $self->zone;
}

=pod

=head3 calculate

Method to calculate the actual shipping cost based on the input (see
methods above). These options can also be passed straight in to this method
(see L<SYNOPSIS>).

Two settings are mandatory: country and weight. The rest are given a
default value that will be used unless told otherwise.

Returns the shipping costs in euro, or undef (see $Business::TNTPost::NL::ERROR
in that case).

=cut

sub calculate {
    my ( $self, %opt ) = @_;

    # Set the options
    for (
        qw/country weight large priority tracktrace
        register receipt machine/
      )
    {
        $self->$_( $opt{$_} ) if ( defined $opt{$_} );
    }

    croak "Not enough information!"
      unless ( defined $self->zone && defined $self->weight );

    # > 2000 grams automatically means 'tracktrace'
    $self->tracktrace(1) if ( $self->weight > 2000 );

    # Zone 1..4 (with tracktrace) automagically means 'priority'
    $self->priority(1) if ( $self->tracktrace );

    # Zone 3,4 + small automagically means 'priority'
    $self->priority(1) if( $self->zone > 3 && !$self->large );

    # Registered (aangetekend) automagically means 'priority'
    #$self->priority(1) if ( $self->register );

    # Fetch the interesting table
    my $ref = _pointer_to_element( table(), $self->_generate_path );
    my $table = $$ref;

    my $highest = 0;
    foreach my $key ( keys %{$table} ) {
        my ( $lo, $hi ) = split ',', $key;
        $highest = $hi if ( $hi > $highest );
        if ( $self->weight >= $lo && $self->weight <= $hi ) {
            $self->cost( $table->{$key} );
            last;
        }
    }
    $ERROR = $self->weight - $highest . " grams too heavy (max: $highest gr.)"
      if ( $highest < $self->weight );

    ### Receipt ("Bericht van ontvangst")
    if ( $self->register && $self->receipt ) {
        if ( $self->zone ) {    # World
            $self->cost( $self->cost + 1.40 );
        }
        else {                  # Netherlands
            $self->cost( $self->cost + 1.15 );
        }
    }

    return ( $self->cost ) ? sprintf( "%0.2f", $self->cost ) : undef;
}

=pod

=head3 _generate_path

Internal method to create the path to walk through the pricing table.
Don't call this, use L<calculate> instead.

=cut

sub _generate_path {
    my $self = shift;

    my @p;

    if ( $self->zone ) {
        push @p, 'world';    # world
        if ( $self->register ) {
            push @p, 'register',    # w/register
              ( $self->zone < 4 )   # w/register/(europe|world)
              ? 'europe'
              : 'world',
              ( $self->machine )   # w/register/(e|w)/(stamp|machine)
                  ? 'machine'
                  : 'stamp';
        }
        elsif ( $self->tracktrace ) {
            push @p, 'plus', 'zone', $self->zone;    # w/plus/zone/[1..4]
        }
        else {
            push @p, 'basic',           # w/basic
              ( $self->zone < 4 )       # w/basic/(europe|world)
              ? 'europe'
              : 'world',
              ( $self->large )		# w/basic/(e|w)/(large|small)
              ? 'large'
              : 'small';

	    if( !$self->large ) {
		# w/basic/(e|w)/small/(machine|stamp)
                push @p, ( $self->machine ) ? 'machine' : 'stamp';
            }

            push @p,
              ( $self->priority )       # w/basic/(e|w)/(l|s)/(m|s)?/(p|s)
              ? 'priority'
              : 'standard';
        }
    }
    else {
        push @p, 'netherlands';                 # netherlands
        if ( $self->register ) {
            push @p, 'register';		# n/register
        }
        else {
            push @p, ( $self->large )           # n/(large|small)
              ? 'large'
              : 'small';
        }
        ( push @p, ( $self->machine ) ? 'machine' : 'stamp' ) 
            unless $self->large;;
    }
#print (join " :: ", @p), "\n";
    return @p;
}

=pod

=head3 _pointer_to_element

Blame L<merlyn> for this internal method. It's using L<List::Util>
to "grep" the information needed.

Don't call this, use L<calculate> instead.

=cut

sub _pointer_to_element {                       # Thanks 'merlyn'!
    require List::Util;
    return List::Util::reduce( sub { \( $$a->{$b} ) }, \shift, @_ );
}

=pod

=head3 weight

Sets and/or returns the weight of the parcel in question in grams.

This value is mandatory for the calculations.

=head3 large

Sets and/or returns the value of this option. Defaults to undef (meaning:
the package will fit through the mail slot).

=head3 priority

Sets and/or returns the value of this option. Defaults to undef (meaning:
standard class (or economy class, where standard is not available)).

=head3 tracktrace

Sets and/or returns the value of this options. Defaults to undef (meaning:
no track & trace feature wanted). When a parcel destined for abroad
weighs over 2 kilograms, default is 1, while over 2kg it's not even
optional anymore.

=head3 register

Sets and/or returns the value of this options. Defaults to undef (meaning:
parcel is not registered (Dutch: aangetekend)).

=head3 receipt

Sets and/or returns the value of this options. Defaults to undef (meaning:
receipt not requested for registered parcels).

=head3 machine

Sets and/or returns the value of this options. Defaults to undef (meaning:
stamps will be used, not the machine (Dutch: frankeermachine)).

Only interesting for destinies within NL. Note that "Pakketzegel AVP"
and "Easystamp" should also use this option.

=head1 BUGS

Please do report bugs/patches to 
L<http://rt.cpan.org/Public/Dist/Display.html?Name=Business-TNTPost-NL>

=head1 CAVEAT

The Dutch postal agency (TNT Post) uses many, many, many various ways
for you to ship your parcels. Some of them are included in this module,
but a lot of them not (maybe in the future? Feel free to patch ;-)

This module handles the following shipping ways (page numbers refer to the 
TNT Post booklet (sorry, all in Dutch)):

=head2 Nederland

=head3 Brievenbuspost

=over 4

=item Brieven, Direct Mail, kaarten, buspakjes en briefkaarten

Pagina 19

=item Aangetekend

Pagina 25, incl. toeslag handtekening retour

=back

=head3 Paketten

=over 4

=item Basis Pakket

Pagina 26

=back

=head2 Buitenland

=head3 Brievenbuspost

=over 4

=item Brieven, Direct Mail, kaarten, buspakjes

Pagina 31

=back

=head3 Pakketten

=over 4

=item Internationaal Pakket Basis

Pagina 35

=item Internationaal Pakket Plus

Pagina 36

=back

=head3 Extra Zeker

=over 4

=item Aangetekend incl. toeslag handtekening retour 

Pagina 32

=back

These should be the most commom methods of shipment.

=head1 AUTHOR

Menno Blom, 
E<lt>blom@cpan.orgE<gt>, 
L<http://menno.b10m.net/perl/>

=head1 COPYRIGHT

This program is free software; you can redistribute
it and/or modify it under the same terms as Perl itself.

The full text of the license can be found in the
LICENSE file included with this module.

=head1 SEE ALSO

L<http://www.tntpost.nl/>, 
L<http://www.tntpost.nl/zakelijk/images/Tarievenboekje_tcm42-411440.pdf>,
L<http://www.iso.org/iso/en/prods-services/iso3166ma/index.html>

=cut

1;
